UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -325,247 +613,406 @@ protected virtual GameObject CreateNode(Node node)
-					mapper.EmissiveTexCoord = def.EmissiveTexture.TexCoord;
+			if (def.OcclusionTexture != null)
+			{
+				mapper.OcclusionTexStrength = def.OcclusionTexture.Strength;
+				int textureId = def.OcclusionTexture.Index.Id;
+				mapper.OcclusionTexture = _assetCache.TextureCache[textureId].Texture;
 
-					//ApplyTextureTransform(def.EmissiveTexture, material, "_EmissionMap");
-				}
+				//ApplyTextureTransform(def.OcclusionTexture, material, "_OcclusionMap");
+			}
 
-				mapper.EmissiveFactor = def.EmissiveFactor.ToUnityColorRaw();
+			if (def.EmissiveTexture != null)
+			{
+				int textureId = def.EmissiveTexture.Index.Id;
+				mapper.EmissiveTexture = _assetCache.TextureCache[textureId].Texture;
+				mapper.EmissiveTexCoord = def.EmissiveTexture.TexCoord;
 
-				var vertColorMapper = mapper.Clone();
-				vertColorMapper.VertexColorsEnabled = true;
+				//ApplyTextureTransform(def.EmissiveTexture, material, "_EmissionMap");
+			}
+
+			mapper.EmissiveFactor = def.EmissiveFactor.ToUnityColorRaw();
+
+			var vertColorMapper = mapper.Clone();
+			vertColorMapper.VertexColorsEnabled = true;
 
-				materialWrapper = new MaterialCacheData
+			MaterialCacheData materialWrapper = new MaterialCacheData
+			{
+				UnityMaterial = mapper.Material,
+				UnityMaterialWithVertexColor = vertColorMapper.Material,
+				GLTFMaterial = def
+			};
+
+			if (materialIndex >= 0)
+			{
+				_assetCache.MaterialCache[materialIndex] = materialWrapper;
+			}
+			else
+			{
+				_defaultLoadedMaterial = materialWrapper;
+			}
+		}
+
+		protected virtual int GetTextureSourceId(GLTF.Schema.Texture texture)
+		{
+			return texture.Source.Id;
+		}
+
+		/// <summary>
+		/// Creates a texture from a glTF texture
+		/// </summary>
+		/// <param name="texture">The texture to load</param>
+		/// <returns>The loaded unity texture</returns>
+		public virtual IEnumerator CreateTexture(GLTF.Schema.Texture texture, int textureIndex, bool markGpuOnly = true)
+		{
+			try
+			{
+				lock (this)
 				{
-					UnityMaterial = mapper.Material,
-					UnityMaterialWithVertexColor = vertColorMapper.Material,
-					GLTFMaterial = def
-				};
+					if (_isRunning)
+					{
+						throw new GLTFLoadException("Cannot CreateTexture while GLTFSceneImporter is already running");
+					}
 
-				if (materialIndex > 0)
+					_isRunning = true;
+				}
+
+				if (_assetCache == null)
+				{
+					InitializeAssetCache();
+				}
+				yield return LoadImageBuffer(texture, GetTextureSourceId(texture));
+				yield return _CreateTexture(texture, textureIndex, markGpuOnly);
+			}
+			finally
+			{
+				lock (this)
 				{
-					_assetCache.MaterialCache[materialIndex] = materialWrapper;
+					_isRunning = false;
 				}
 			}
+		}
+
+		/// <summary>
+		/// Gets texture that has been loaded from CreateTexture
+		/// </summary>
+		/// <param name="textureIndex">The texture to get</param>
+		/// <returns>Created texture</returns>
+		public virtual UnityEngine.Texture GetTexture(int textureIndex)
```

Comment:
This function has zero references. Can we remove it?

Reply:
We use it in our code. It's the only way to get the result from CreateTexture if called publically

Reply:
A better way to do this is probably to match the calling convention of the other functions and access a public member